quinn-proto/src/crypto.rs

```diff
@@ -135,6 +121,32 @@ pub trait CryptoSession {
     fn write_handshake(&mut self, buf: &mut Vec<u8>) -> Result<Option<Crypto>, TLSError>;
 }
 
+impl CryptoClientConfig for Arc<ClientConfig> {
```

Comment:
Why for `Arc<ClientConfig>` instead of for `ClientConfig`?

Reply:
With just `ClientConfig` the compiler doesn't accept it. Here too, I'd prefer to leave it slightly ugly for now until the rest of the picture about this abstraction becomes more clear (for example, in this case I think we might want to pull in the `ClientConfig` defined in the quinn crate and define the `impl` on that).